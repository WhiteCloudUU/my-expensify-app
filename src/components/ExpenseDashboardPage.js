import React from 'react';
import { Link } from 'react-router-dom';
import ExpenseList from './ExpenseList'
import ExpenseListFilters from './ExpenseListFilters'

const ExpenseDashboardPage = () => (
    <div>
        Expense dashboard

        <Link to='edit/42'>Expense 42</Link>
        <Link to='edit/1024'>Expense 1024</Link>
        <ExpenseListFilters />
        <ExpenseList />
    </div>

);

export default ExpenseDashboardPage;