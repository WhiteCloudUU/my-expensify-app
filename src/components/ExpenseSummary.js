import React from 'react'
import numeral from 'numeral'
import { connect } from 'react-redux'
import getExpenseTotal from '../selectors/expenses-total'
import selectExpenses from '../selectors/expenses'


export const ExpenseSummary = ({expensesCount, expensesTotal}) => {
    
    return (
        <div>
            <h1>
                Viewing {expensesCount} expenses, 
                totaling at {numeral(expensesTotal / 100).format('$0,00.00')}.
            </h1>
        </div>
    )
};

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);

    return {
        expensesCount: visibleExpenses.length,
        expensesTotal: getExpenseTotal(visibleExpenses)
    }
};
    


export default connect(mapStateToProps)(ExpenseSummary)
