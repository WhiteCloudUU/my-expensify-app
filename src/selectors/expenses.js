
import moment from 'moment'

export default (expenses, { text, sortBy, startDate, endDate }) => {
    return expenses.filter((expense) => {
        const createdAtMoment = moment(expense.createdAt);
        const startDateMatch = startDate ? createdAtMoment.isSameOrAfter(startDate, 'day'): true;
        const endDateMatch = endDate ? createdAtMoment.isSameOrBefore(endDate, 'day') : true;
        const textMatch = expense.description.toLowerCase().includes(text.toLowerCase());
        
        return textMatch && startDateMatch && endDateMatch;

    }).sort((a, b) => {
        if (sortBy == "date") {
            return a.createdAt < b.createdAt ? 1 : -1;
        } else if (sortBy == "amount") {
            return a.amount < b.amount ? 1 : -1;
        }
    })
}