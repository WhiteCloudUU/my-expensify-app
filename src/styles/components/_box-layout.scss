.box-layout {
    background: url('/images/bg.jpg');
    background-size: cover;
    height: 100vh;
    width: 100wh;
    display: flex;
    align-items: center;
    justify-content: center;
}

.box-layout__box {
    background: fade-out($color: white, $amount: .15);
    border-radius: 10px;
    padding: $l-size $m-size;
    text-align: center;
    width: 25rem;
}

.box-layout__title {
    margin-bottom: $m-size;
    line-height: 1;
}