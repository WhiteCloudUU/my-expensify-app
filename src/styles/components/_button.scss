.button {
    background: $dark-blue;
    color: white;
    display: inline-block;
    font-weight: 300;
    font-size: $large-font;
    line-height: 1;
    border: none;
    border-radius: 5px;
    padding: $s-size;
    text-decoration: none;
}

.button__link {
    background: none;
}

.button--second {
    background: #888888;
}

