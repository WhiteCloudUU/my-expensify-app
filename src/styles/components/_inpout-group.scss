.input-group {
    display: flex;
    flex-direction: column;
    margin-bottom: $m-size;
    @media (min-width: $desktop-breakpoint) {
        flex-direction: row;
        margin-bottom: $l-size;
    }
    
}

.input-group__item {
    margin: 0 0 $s-size 0; // set margin-bottom in mobile app
    @media (min-width: $desktop-breakpoint) {
        margin: 0 $s-size 0 0; // set margin-right in desktop app
    }
}