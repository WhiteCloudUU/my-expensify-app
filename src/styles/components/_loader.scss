.loader {
    display: flex;
    align-items: center;
    justify-content: center;
    height: 100vh;
    width: 100wh;
}

.loader-image {
    height: 10rem;
    width: 10rem;
}