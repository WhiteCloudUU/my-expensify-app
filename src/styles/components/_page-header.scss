.page-header {
    background: $off-white;
    margin: 0 0 $l-size 0;
    padding: $l-size 0;
}

.page-header__title {
    font-weight: 300;
    margin: 0;
    span {
        font-weight: 700;
    }
}

.page-header__action {
    margin-top: $m-size;
}



