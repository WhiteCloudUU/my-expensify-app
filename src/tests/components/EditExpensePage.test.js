import React from 'react'
import { shallow } from 'enzyme'
import { EditExpensePage } from '../../components/EditExpensePage'
import expenses from '../fixtures/expenses'

let editExpenseSpy, removeExpenseSpy, history, wrapper;

beforeEach(() => {
    editExpenseSpy = jest.fn();
    removeExpenseSpy = jest.fn();
    history = { push: jest.fn() };
    wrapper = shallow(<EditExpensePage 
                        expense={expenses[0]}
                        editExpense={editExpenseSpy} 
                        removeExpense={removeExpenseSpy}
                        history={history} />);
});

test('should render EditExpensePage correctly', () => {
    expect(wrapper).toMatchSnapshot();
})

test('should handle edit expense', () => {
    wrapper.find('ExpenseForm').prop('onSubmit')(expenses[0]);
    expect(history.push).toHaveBeenLastCalledWith('/');
    expect(editExpenseSpy).toHaveBeenLastCalledWith(expenses[0].id, expenses[0]);
})

test('should handle remove expense', () => {
    wrapper.find('button').simulate('click');
    expect(history.push).toHaveBeenLastCalledWith('/');
    expect(removeExpenseSpy).toHaveBeenLastCalledWith(expenses[0].id);
})

