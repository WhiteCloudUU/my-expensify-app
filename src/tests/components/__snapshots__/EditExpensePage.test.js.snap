// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render EditExpensePage correctly 1`] = `
<div>
  <ExpenseForm
    expense={
      Object {
        "amount": 475,
        "createdAt": 0,
        "description": "Coffee",
        "id": 1,
        "note": "",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
